pkg/rc/replication_controller.go

```diff
@@ -893,9 +896,14 @@ func (rc *replicationController) watchHealth() bool {
 	for {
 		select {
 		case err := <-errCh:
```

Comment:
One possible reason we saw a nil error here might be that `errCh` closed, you can check that by doing `err, ok := <-errCh` and then checking `ok`. It will be false if the channel was closed